bindings/rust/s2n-tls/src/client_hello.rs

```diff
@@ -206,6 +132,95 @@ impl ClientHello {
     }
 }
 
+#[cfg(feature = "unstable-fingerprint")]
+pub use self::fingerprint::*;
+
+// This module is public in the client_hello crate, but can only
+// be accessed by users by enabling the unstable-fingerprint feature.
+// This can be removed when the fingerprint feature has become stable.
+pub(super) mod fingerprint {
+    use crate::error::{Error, Fallible};
+    use s2n_tls_sys::*;
+
+    use super::ClientHello;
+
+    #[derive(Copy, Clone)]
+    pub enum FingerprintType {
+        JA3,
+    }
+
+    // this is the size of the MD5 hash digest that is used for the JA3 fingerprint
+    pub const MD5_HASH_SIZE: u32 = 16;
+
+    impl From<FingerprintType> for s2n_tls_sys::s2n_fingerprint_type::Type {
+        fn from(value: FingerprintType) -> Self {
+            match value {
+                FingerprintType::JA3 => s2n_tls_sys::s2n_fingerprint_type::FINGERPRINT_JA3,
+            }
+        }
+    }
+
+    impl ClientHello {
+        /// `fingerprint_hash` calculates the hash, and also returns the size
+        /// required for the full fingerprint string. The return value can be used
+        /// to construct a string of appropriate capacity to call
+        /// `fingerprint_string`. `output` will be extended if necessary to store
+        /// the full hash.
+        pub fn fingerprint_hash(
```

Comment:
This isn't how visibility works. You will end up exporting these regardless of the feature being enabled.

Reply:
Yeah it's weird. The fingerprint_hash function gets exported, but you can't actually call it because the FingerprintType parameter is private. So... technically the function is not callable xD. I dunno, it's totally a hack.

Reply:
That's not great UX. Why are we not just gating these methods to the feature?

Reply:
I reverted it back. I agree, it's bad UX and I don't think it's worth it. The original code had the hash in the debug fmt implementation, so I was trying to preserve that ability. But I don't think there is a clean way to do so.